docs/src/content/en/docs/workflows/human-in-the-loop.mdx

```diff
@@ -1,268 +1,126 @@
-- Answer yes/no questions with clear "Yes" or "No" responses
-- Be consistent - same question asked differently should get the same answer
-- Ask for clarification if a question is unclear
-- If multiple questions are asked at once, ask them to ask one at a time
-
-When they make a guess:
-- If correct: Congratulate them warmly
-- If incorrect: Politely correct them and encourage them to try again
-
-Encourage players to make a guess when they seem to have enough information.
-
-You must return a JSON object with:
-- response: Your response to the user
-- gameWon: true if they guessed correctly, false otherwise`,
-  model: openai("gpt-4o"),
-});
-```
+## Requesting human approval with `suspend()`
 
-## Multi-turn workflow
+Use `suspend()` to pause workflow execution at a specific step while waiting for human approval. If a `resumeSchema` condition isn’t met, the workflow suspends at `return suspend()`. No further logic runs, and only the values validated against the optional `suspendSchema` and passed to `suspend()` are returned. The step’s main return block is skipped, and workflow execution doesn’t continue.
 
-The workflow coordinates the full interaction using `suspend`/`resume` to pause for human input and `dountil` to repeat the game loop until a condition is met.
+![Pausing a workflow with suspend()](/img/workflows/workflows-suspend-resume-suspend.jpg)
 
-The `startStep` generates a name using the `famousPersonAgent`, while the `gameStep` runs the interaction through the `gameAgent`, which handles both questions and guesses and produces structured output that includes a `gameWon` boolean.
-
-```typescript title="src/mastra/workflows/example-heads-up-workflow.ts" showLineNumbers copy
+```typescript {12-17,22-26} title="src/mastra/workflows/test-workflow.ts" showLineNumbers copy
 import { createWorkflow, createStep } from "@mastra/core/workflows";
 import { z } from "zod";
 
-const startStep = createStep({
-  id: "start-step",
-  description: "Get the name of a famous person",
+const step1 = createStep({
+  id: "step-1",
   inputSchema: z.object({
-    start: z.boolean(),
+    userEmail: z.string()
   }),
   outputSchema: z.object({
-    famousPerson: z.string(),
-    guessCount: z.number(),
-  }),
-  execute: async ({ mastra }) => {
-    const agent = mastra.getAgent("famousPersonAgent");
-    const response = await agent.generate("Generate a famous person's name", {
-      temperature: 1.2,
-      topP: 0.9,
-      memory: {
-        resource: "heads-up-game",
-        thread: "famous-person-generator",
-      },
-    });
-    const famousPerson = response.text.trim();
-    return { famousPerson, guessCount: 0 };
-  },
-});
-
-const gameStep = createStep({
-  id: "game-step",
-  description: "Handles the question-answer-continue loop",
-  inputSchema: z.object({
-    famousPerson: z.string(),
-    guessCount: z.number(),
+    output: z.string()
   }),
   resumeSchema: z.object({
-    userMessage: z.string(),
+    okToSend: z.boolean()
```

Comment:
i generally appreciate literal identifiers that feel friendly like `okToSend` but tbh, `approved` is probably more colloquial here

Reply:
this is basically human-in-the-loop btw and the relationship between these two pages is potentially confusing 

if we have a dedicated page on hitl, it stands to reason this page is about sleep & events, but we have no example code for that